js/models/questionModel.js

```diff
@@ -291,18 +291,22 @@ class QuestionModel extends ComponentModel {
       case 'partlyCorrect': {
         if (typeof _feedback._partlyCorrect === 'object') {
           // old style
+          const fallbackBody = _feedback._partlyCorrect?.final || _feedback._incorrect?.final || ''
+          const body = !isFinal ? _feedback._partlyCorrect?.notFinal || fallbackBody : fallbackBody;
+
           return {
             // add higher values
             altTitle,
             title,
             _classes,
-            body: !isFinal
-              ? _feedback._partlyCorrect?.notFinal || _feedback._incorrect?.notFinal || ''
-              : _feedback._partlyCorrect?.final || _feedback._incorrect?.final || ''
+            body
           };
         }
+
         // new style
-        const feedbackPartlyCorrect = !isFinal ? _feedback._partlyCorrectNotFinal || _feedback._incorrectNotFinal : _feedback._partlyCorrectFinal || _feedback._incorrectFinal;
+        const fallbackFeedback = _feedback._partlyCorrectFinal || _feedback._incorrectFinal || ''
+        const feedbackPartlyCorrect = !isFinal ? _feedback._partlyCorrectNotFinal || fallbackFeedback : fallbackFeedback;
```

Comment:
```suggestion
        const finalFeedback = _feedback._partlyCorrectFinal || _feedback._incorrectFinal || '';
        const notFinalFeedback = _feedback._partlyCorrectNotFinal || finalFeedback;
        const feedbackPartlyCorrect = isFinal 
          ? finalFallbackFeedback 
          : notFinalFeedback;
```
Could you do similar to the others? Isolate the oddities so that they're easier to read.

Reply:
@oliverfoster I actually ended up refactoring the feedback object building a little bit to reduce code repetition. Could you take a look please and see what you think?
Is the "old style" _feedback.correct the right structure? If this was a typo and there was a missing underscore (i.e. it should be `_feedback._correct` then further simplification would be possible on that switch statement

Reply:
Old style: https://github.com/adaptlearning/adapt_framework/blob/21f28212cf8e5635fe7f71d09279917861b9a3f3/src/course/en/components.json#L450-L458

New style: https://github.com/adaptlearning/adapt-contrib-core/pull/223

Reply:
You don't seem to have eslint in your editor, or it's not working. I have missing semi-colons on line 304, 313, 315, 320